### PR TITLE
upgrading tests to JUnit 5

### PR DESCRIPTION
**Pull request by elbiocaetano**
1. sda-commons-server-kafka
2. sda-commons-server-mongo-testing
3. sda-commons-server-openapi

### CODE REVIEW

File: sda-commons-server-kafka/src/test/java/com/salesforce/kafka/test/SharedKafkaClassExtensionScram.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  /**
+   * Calls a command that sets up SCRAM-SHA-* credentials in the zookeeper. There is no other
+   * possibility to do this yet (see also
+   * https://cwiki.apache.org/confluence/display/KAFKA/KIP-506%3A+Allow+setting+SCRAM+password+via+Admin+interface).
+   *
+   * @param username the name of the user to update
+   * @param password the password to set
+   */
+  public void configureScramUser(String username, String password) {
```
Comment (christopher-cudennec) [inline sda-commons-server-kafka/src/test/java/com/salesforce/kafka/test/SharedKafkaClassExtensionScram.java:75]: Let's try to make this method `private`.

File: sda-commons-server-kafka/src/test/java/com/salesforce/kafka/test/SharedKafkaClassExtensionScram.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  @Override
+  public void beforeAll(ExtensionContext context) throws Exception {
+    // First start Zookeeper to be able to change the configuration
+    sharedZookeeperTestResource.beforeAll(context);
+    configureScramUser("admin", "admin-secret");
+    configureRegisteredUser();
+    super.beforeAll(context);
+  }
+
+  private void configureRegisteredUser() {
```
Comment (christopher-cudennec) [inline sda-commons-server-kafka/src/test/java/com/salesforce/kafka/test/SharedKafkaClassExtensionScram.java:51]: I would move this method down for better ordering (next to `configureScramUser`)

File: sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbRuleTest.java [hunk 3]
```diff
@@ -5,16 +5,17 @@
...
+import org.junit.jupiter.api.Test;
 import org.junit.runner.Description;
 import org.junit.runners.model.Statement;
 import org.sdase.commons.server.testing.SystemPropertyRule;
 
-public class UseExistingMongoDbRuleTest {
+class UseExistingMongoDbRuleTest {
 
   @ClassRule
   public static final MongoDbRule EXTERNAL_DB =
```
Comment (christopher-cudennec) [inline sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbRuleTest.java:21]: Keeping the old JUnit 4 annotation looks odd. I'd prefer reverting the test to the old JUnit 4 state. We will remove it when we remove all our provided JUnit 4 rules.

File: sda-commons-server-openapi/src/test/java/org/sdase/commons/server/openapi/OpenApiBundleDefaultPortIT.java [hunk 4]
```diff
@@ -9,21 +9,21 @@
...
 import org.sdase.commons.server.openapi.test.OpenApiAssertions;
 
-@Ignore(
+@Disabled(
     "This test can not be run on the build server since it binds to port 80 which is not permitted (Permission denied)")
-public class OpenApiBundleDefaultPortIT {
-  @ClassRule
-  public static final DropwizardAppRule<Configuration> DW =
-      new DropwizardAppRule<>(
+class OpenApiBundleDefaultPortIT {
```
Comment (christopher-cudennec) [inline sda-commons-server-openapi/src/test/java/org/sdase/commons/server/openapi/OpenApiBundleDefaultPortIT.java:23]: Strange test. I guess it was not executed for a couple of years. @JoergSiebahn Can you remember the purpose of the test?

Comment (JoergSiebahn) [reply sda-commons-server-openapi/src/test/java/org/sdase/commons/server/openapi/OpenApiBundleDefaultPortIT.java:23]: Maybe it's about to verify that the port is [not part of the server list](https://github.com/SDA-SE/sda-dropwizard-commons/pull/1952/files#diff-0769157ed7fcff783f6a77e2b0ebf927def64e94247d29548b0e0c08d7482820R58)?

File: sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbRuleTest.java [hunk 5]
```diff
@@ -24,12 +25,13 @@ public class UseExistingMongoDbRuleTest {
...
           .build();
 
-  private static SystemPropertyRule SYSTEM_PROPERTY_RULE;
+  @ClassRule public static SystemPropertyRule SYSTEM_PROPERTY_RULE;
 
   private MongoDbRule useExistingMongoDbRule;
 
   @BeforeClass
   public static void initEnvAfterStartOfExternalDb() {
+    ((StartLocalMongoDbRule) EXTERNAL_DB).before();
```
Comment (christopher-cudennec) [inline sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbRuleTest.java:34]: Why do we need to invoke the method manually? EXTERNAL_DB is also defined as `ClassRule`

File: sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbRuleTest.java [hunk 6]
```diff
@@ -66,6 +67,11 @@ public void clearDatabase() {
     useExistingMongoDbRule.clearDatabase();
   }
 
+  @AfterClass
+  public static void afterClass() {
+    ((StartLocalMongoDbRule) EXTERNAL_DB).after();
+  }
```
Comment (christopher-cudennec) [inline sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbRuleTest.java:73]: This should also be covered by the `ClassRule`.

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**